cx/spark/server.js

```diff
@@ -0,0 +1,208 @@
+// const targetUrl = 'https://example.com';
+// const projectId = 'my-project';
+// const locationId = 'global';
+// const agentId = 'my-agent';
+// const languageCode = 'en'
+
+// Imports the Google Cloud Some API library
+const {SessionsClient} = require('@google-cloud/dialogflow-cx');
+/**
+ * Example for regional endpoint:
+ *   const locationId = 'us-central1'
+ *   const client = new SessionsClient({apiEndpoint: 'us-central1-dialogflow.googleapis.com'})
+ */
+const client = new SessionsClient({apiEndpoint: locationId + '-dialogflow.googleapis.com'});
+
+// Upon start a webhook is registered with spark
+// Upon closure the webhook is removed from spark
+
+const listener = app.listen(process.env.PORT, async function() {
+  await init();
+  console.log('Your Spark integration server is listening on port '
+      + listener.address().port);
+});
+
+app.post('/', async function(req, res) {
+  const message = await retrieveMessage(req.body.data.id);
+  const dialogflowResponse = await detectIntentText(message, req.body.data.personId);
+  const sparkMessage = detectIntentToSparkMessage(dialogflowResponse);
+  sendMessage(sparkMessage.text, message.email);
+});
+
+process.on('SIGTERM', () => {
+  listener.close(async ()=>{
+    console.log('Closing http server.');
+    await deleteWebhooks();
+    process.exit(0);
+  });
+});
+
+async function init(){
+  await deleteWebhooks();
+  registerWebhook();
+}
+
+// Converts Spark message to a detectIntent request. 
+function sparkToDetectIntent(message, sessionPath){
+    const request = {
+        session: sessionPath,
+        queryInput: {
+            text: {
+                text: message.text,
+            },
+            languageCode,
+        },
+    };
+
+  return request;
+}
+
+// Converts detectIntent response to a Spark text message. 
+function detectIntentToSparkMessage(response){
+    agentResponse = '';
+    
+    for (const message of response.queryResult.responseMessages) {
+        if (message.text) {
+            agentResponse += `${message.text.text}\n`;
+        };
+    };
+  
+  if(agentResponse.length != ''){
+        const request = {
+            text: agentResponse
+        };
+        return request;
+    };
+};
+
+/**
+ * This function calls Dialogflow CX API to retrieve the response
+ * https://cloud.google.com/dialogflow/cx/docs/quick/api
+ */
+async function detectIntentText(message, personId) {
+  const sessionId = personId;
+  const sessionPath = client.projectLocationAgentSessionPath(
+      projectId,
+      locationId,
+      agentId,
+      sessionId
+  );
+  console.info(sessionPath);
+
+  request = sparkToDetectIntent(message, sessionPath);
+  const response = await client.detectIntent(request);
+
+  return response;
+}
+
+function sendMessage(message, personEmail) {
+  request.post('https://api.ciscospark.com/v1/messages', {
+    auth: {
+      bearer: sparkAccessToken
+    },
+    json: {
+      "toPersonEmail": personEmail,
+      "text": message.text
```

Comment:
It might be better if this function could provide the flexibility for sending non-text message (which is the purpose of changing function interface).

Reply:
The new change is still limited to text message. Maybe consider something like:
```
json: JSON.stringify(message)
```